test/test_coordgen.cpp

```diff
@@ -0,0 +1,43 @@
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE Test_Coordgen
+
+#include <boost/filesystem.hpp>
+#include <boost/test/unit_test.hpp>
+
+#include "../sketcherMinimizer.h"
+
+#include "maeparser/MaeConstants.hpp"
+#include "maeparser/Reader.hpp"
+
+using namespace schrodinger;
+
+const boost::filesystem::path test_samples_path(TEST_SAMPLES_PATH);
+
+BOOST_AUTO_TEST_CASE(SampleTest)
+{
+    // A small sample test showing how to import a molecule from a .mae file and
+    // initialize the minimizer with it.
+
+    const std::string testfile = (test_samples_path / "test.mae").string();
+
+    mae::Reader r(testfile);
+    auto block = r.next(mae::CT_BLOCK);
+    BOOST_REQUIRE(block != nullptr);
+
+    auto* mol = mol_from_mae_block(*block);
+    BOOST_REQUIRE(mol != nullptr);
+    BOOST_REQUIRE_EQUAL(mol->getAtoms().size(), 26);
+    BOOST_REQUIRE_EQUAL(mol->getBonds().size(), 26);
+
+    sketcherMinimizer minimizer;
+    minimizer.initialize(mol); // minimizer takes ownership of mol
```

Comment:
Thanks for the comment.

Reply:
No problem. I was quite puzzled when I attempted to clean up mol and valgrind returned an error. Then I found out that the minimizer takes ownership and cleans it up on destruction.